elastalert/test_rule.py

```diff
@@ -271,18 +267,21 @@ def run_rule_test(self):
         parser.add_argument('--count-only', action='store_true', dest='count', help='Only display the number of documents matching the filter')
         args = parser.parse_args()
 
+        with open(args.file) as fh:
+            rule_yaml = yaml.load(fh)
+
         if args.json:
             with open(args.json, 'r') as data_file:
                 self.data = simplejson.loads(data_file.read())
         else:
-            hits = self.test_file(args)
+            hits = self.test_file(copy.deepcopy(rule_yaml), args)
```

Comment:
Validation is not idempotent, since it mutates the rule object, changing, for example, `{"hours": 4}` to a `datetime.timedelta`. Therefore, we have to create a deepcopy if we want to run the rule through validation twice.